### PR TITLE
Initial organisation (and user) management flows

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
Building upon #11 and #14, this PR adds the following features to the prototype:

* organisations (with parent and child relationships)
* views for CORE administrators
* deletion as well as deactivation of users

This PR **does not** add:

* Adding/editing organisation details
* Ability to reassign logs
* Ability to reassign logs when deactivating or deleting a user
* Ability to filter/manage large amounts of data – we can address that once we are happy with the foundations and general approach

## CORE administrators

* Log in with the following email `admin@communities.gov.uk` (no password needed)
* 2 additional items are shown in the header navigation: ‘Organisations’ and ‘Users’

### Organisations 

An admin can view all organisations:

admin-organisations

An admin can deactivate an organisation, and change any of its properties:

admin-organisation

An admin can also view users scoped to a particular organisation

admin-organisation-users

**NOTE:** This view currently shows all users *related* to an organisation (ie this organisation and users at any child organisations). This makes sense when viewed from the perspective of a data coordinator (see later screens), but perhaps not from the perspective of a CORE administrator. 🤔

### Users 

An admin can view all users across all organisations:

admin-users

An admin can deactivate or delete a user, and change any of their account details:

admin-user

## Data coordinator at an owning (aka parent) organisation

* Log in with the following email `data.coordinator@owning.org.uk` (no password needed)
* One additional item is shown in the header navigation: ‘Your organisation’

### Your organisation

parent-coordinator-organisation

**NOTE:** Rather than say parent and child in the UI, I have referred to parents as ‘Stock owners’ and children as ‘Managing agents’. These terms are probably wrong, but we should aim to use terminology that is accurate and understandable – it may even turn out that parent/child are the best understood terms!

### Your organisation → Users

parent-coordinator-users

### Managing logs

*Lots* more to do here, but updated this view to show actions a data coordinator may be able to perform (that a data provider wouldn’t)

parent-coordinator-logs

### Inviting a new user

As well as asking for personal details and role, we now ask which organisation a user should be associated with. We can also show the organisation’s name when showing the different permissions a role gives. 

parent-coordinator-new-user-details

parent-coordinator-new-user-organisation

parent-coordinator-new-user-role

**NOTE**: The prototype currently allows you to add a user to any organisation, which is incorrect. Future iterations can ensure we only show the organisations a data coordinator has permissions to manage. And we’d likely not use a dropdown menu, either.

**NOTE**: These role descriptions are super bad, but hopefully get the idea across!

## Data coordinator at an managing (aka child) organisation

* Log in with the following email `data.coordinator@managing.org.uk` (no password needed)
* One additional item is shown in the header navigation: ‘Your organisation’

### Your organisation

child-coordinator-organisation

### Your organisation → Users

While the views are the same as that seen by a data coordinator at a parent organisation, less information is shown. For example, the list of users shows only those in the child organisation:

child-coordinator-users

## Data provider at an owning (aka parent) or managing (aka child) organisation

* Log in with either of the following emails:  `data.provider@owning.org.uk` or `data.provider@managing.org.uk` (no passwords needed)
* No additional items are shown in the header navigation. The journey is pretty much as described in #11, except you can now see which organisation you belong to in your user profile.

### CODE REVIEW

File: app/data/questions/local-authorities.js [hunk 1]
```diff
@@ -1,1534 +1,18 @@
...
+    const fb = b.name.toLowerCase()
 
     if (fa < fb) { return -1 }
     if (fa > fb) { return 1 }
     return 0
   })
+  .map(la => ({ text: la.name, value: la.gss }))
 
 localAuthoritiesInEngland.unshift({
-  UID: 0,
```
Comment (roobottom) [inline app/data/questions/local-authorities.js:1531]: I mean... This is just anal. ;-)

File: app/data/questions/local-authorities.js [hunk 2]
```diff
@@ -1,1534 +1,18 @@
...
-}, {
-  UID: 378,
-  value: 'S12000010',
-  text: 'East Lothian'
-}, {
-  UID: 379,
-  value: 'S12000011',
-  text: 'East Renfrewshire'
-}]
+import localAuthorities from '../../datasets/local-authorities.js'
```
Comment (roobottom) [inline app/data/questions/local-authorities.js:1]: Why have you moved some of this to datasets? Is it used in multiple places now?

File: app/data/questions/yes-no-postcode.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ export default [{
   text: 'Yes',
   value: 'true',
   conditional: {
-    html: `<p>Postcode question goes here</p>`
+    html: '<p>Postcode question goes here</p>'
```
Comment (roobottom) [inline app/data/questions/yes-no-postcode.js:5]: Good spot - but this entire file isn't used. I'll just let that sit with your OCD for a while and see what happens. ;-)

**Review by roobottom — APPROVED** [review]
LGTM

File: app/data/questions/yes-no-postcode.js (hunk 3 above)
Comment (paulrobertlloyd) [reply app/data/questions/yes-no-postcode.js:5]: I can’t even remember doing this… but right, it’s going in the 🗑️!

File: app/data/questions/local-authorities.js (hunk 1 above)
Comment (paulrobertlloyd) [reply app/data/questions/local-authorities.js:1531]: 👋

**Pull request merged**